### PR TITLE
Multiple Light Source handling on players + Light Cones

### LINKED ISSUE #6722: Lighting does not update after you activate more powerful light source
### Environment
- Game Version: 21060314

### Detailed description of bug

Flashlight is more powerful light source than PDA, but if PDA light is on, emitted light stays on same level if you turn flashlight on/off.
This applies to all light sources with varying brightness and color (color stays same until you turn off previous light source)

Demonstration:

https://user-images.githubusercontent.com/22667809/120896258-7d1ddd00-c629-11eb-964b-5c110a17b4fa.mp4

### Steps to reproduce

1. turn on PDA light
2. turn on flashlight light

### LINKED ISSUE #6780: Light level differs significantly between equipped/held items and items on the ground
### Environment
- Game Version: 21060821

### Detailed description of bug

PDA in hands:
image

PDA on ground (dimmer):
image

firefighter helmet in hands:
image

firefighter helmet on ground (brighter):
image

### PR DESCRIPTION
**Pull request by MaxIsJoe**
image

fixes #6722
fixes #6780

### Changelog:

CL: [New] Added directional lights for flashlights (Including the PDA's flashlight).
CL: [Improvement] Players can now have several light sources emitting on them instead of having one light source only.
CL: [Improvement] Items that emit light previously would have inconsistencies with strength when it comes to being held or while on the ground. This behavior was understandably unintentional, but it is now a proper setting for mapmakers / designers to use with tweakable settings. As of right now, this feature is only relegated towards small items like penlights.
CL: [Improvement] Several light items have had their colors and strengths adjusted to be easier on the eye and provide more visibility.
CL: [Improvement] The ALR's light sprite has been replaced with a more unique shape.
CL: [Fix] Items that emit light will now correctly show their pre-defined settings regardless if they're held in-hand or are on the ground.
CL: [Fix] Fixed an issue with the ALR own light getting stuck on the ground when picked up by a player.

### CODE REVIEW

**Comment by corp-0** [comment]
Give a look to the following issues and see if your PR closes them.

- #6780 
- #6722

**Comment by MaxIsJoe** [comment]
The bottom one does get fixed 
Ill test the top one when im back home

**Comment by MaxIsJoe** [comment]
I think I've caught and fixed every bug I could find, once tests pass; pls merge.

File: UnityProject/Assets/Scripts/Core/Lighting/LightsHolder.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+using System.Linq;
+using Light2D;
+using Mirror;
+using UnityEngine;
+
+namespace Core.Lighting
+{
+	public class LightsHolder : NetworkBehaviour
+	{
+		public SyncList<LightData> Lights { get; private set; } = new SyncList<LightData>();
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Lighting/LightsHolder.cs]: You got a mark is dirty when adding and removing or modifying

File: UnityProject/Assets/Scripts/Core/Lighting/LightsHolder.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			foreach (var sprite in lightsParent.GetComponentsInChildren<LightSprite>())
+			{
+				if (sprite.GivenID == data.Id) Despawn.ClientSingle(sprite.gameObject);
+			}
+
+			Lights.Remove(data);
+			UpdateLights();
+		}
+
+		[ClientRpc]
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Lighting/LightsHolder.cs]: there's a hook for  SyncList so you don't need to manually call, and also re-logging works with hook

File: UnityProject/Assets/Scripts/Core/Lighting System/Light2D/LightSprite.cs [hunk 3]
```diff
@@ -25,6 +25,8 @@ public class LightSprite : CustomSprite
         private Vector3 _oldLightOrigin;
         private LightShape _oldLightShape;
 
+        public int GivenID = 0;
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Lighting System/Light2D/LightSprite.cs:28]: might be better to make a system to handle spawning in any sprite, using sprite handler  system,  that a lot of work so it's okay just to do it for this

**Review by corp-0 — APPROVED** [review]

**Pull request merged**